subcommand/common/common.go

```diff
@@ -30,3 +32,17 @@ func Logger(level string) (hclog.Logger, error) {
 		Output: os.Stderr,
 	}), nil
 }
+
+// ValidatePort converts flags representing ports into integer and validates
+// that it's in the port range.
+func ValidatePort(flagName, flagValue string) error {
+	port, err := strconv.Atoi(flagValue)
+	if err != nil {
+		return errors.New(fmt.Sprintf("%s value of %s is not a valid integer.", flagName, flagValue))
+	}
+	// This checks if the port is in the valid port range.
+	if port < 1024 || port > 65535 {
```

Comment:
Ports < 1024 are valid I think.

Reply:
Is this checking for unprivileged ports?

Reply:
Yes that's accurate, sorry for the imprecise wording. It's checking for unprivileged ports and will return an error if you configure something in the unprivileged port range. I can make a change to the comment to clarify.

Reply:
Is there a possible case where a user exposes their prom metrics on a privileged port because they're running their container as root? In that case we only want to ensure that the metrics merged port is unprivileged and valid and then we want to check that the scrape port is simply valid but it can be privileged or unprivileged.